core/domain/topic_services.py

```diff
@@ -159,9 +159,10 @@ def _create_topic(committer_id, topic, commit_message, commit_cmds):
         committer_id: str. ID of the committer.
         topic: Topic. topic domain object.
         commit_message: str. A description of changes made to the topic.
-        commit_cmds: list(dict). A list of change commands made to the given
-            topic.
+        commit_cmds: list(TopicChange). A list of TopicChange objects that
```

Comment:
This change looks great. Would it be hard to do something similar (just exactly these changes) to explorations and collections? We can save the "get rid of the dicts entirely" refactor to later.

If it's hard, no worries, and it's OK to file an issue for those instead. But it would be nice to have our codebase be reasonably consistent so that there aren't two different patterns being used simultaneously (and so that, e.g., commit_cmds always refers to a list of domain objects when that term is used throughout the codebase).

Reply:
Yeah, that's fine. I'll do the same changes for explorations and collections as well.

Reply:
Done, added for explorations here, will do collections in another PR.